src/System.Net.Security/src/System/Net/Security/SslState.cs

```diff
@@ -1279,18 +1280,39 @@ internal void FinishRead(byte[] renegotiateBuffer)
             }
         }
 
-        // Returns: 
-        // true  - operation queued
-        // false - operation can proceed
-        internal bool CheckEnqueueWrite(AsyncProtocolRequest asyncRequest)
+        internal Task CheckEnqueueWriteAsync()
+        {
+            //Clear previous request.
+            int lockState = Interlocked.CompareExchange(ref _lockWriteState, LockWrite, LockNone);
+            if (lockState != LockHandshake)
+            {
+                return Task.CompletedTask;
+            }
+
+            lock (this)
+            {
+                if (_lockWriteState != LockHandshake)
+                {
+                    CheckThrow(true);
```

Comment:
Nit: can we name the argument in the call?  I don't know what "true" means here.